File: lib/db/mongo.py
from lib.settings import Settings
from pymongo import MongoClient
from pymongo.collection import Collection
from pymongo.database import Database
from typing import Optional

instance: Optional[MongoClient] = None


def get_client() -> MongoClient:
    global instance
    if instance is None:
        kwargs = {}
        if Settings.mongo_db_user and Settings.mongo_db_password and Settings.mongo_db_auth_db:
            kwargs['username'] = Settings.mongo_db_user
            kwargs['password'] = Settings.mongo_db_password
            kwargs['authSource'] = Settings.mongo_db_auth_db
        instance = MongoClient(Settings.mongo_db_host, Settings.mongo_db_port, **kwargs)
    return instance


def get_database(db=Settings.mongo_db_db_name) -> Database:
    client = get_client()
    return client[db]


def get_collection(name, db=Settings.mongo_db_db_name) -> Collection:
    client = get_client()
    return client[db][name]

File: MoonlightDashboard/MoonlightDashboard/Db/Mongo.cs
using MongoDB.Bson;
using MongoDB.Driver;
using MoonlightDashboard.Lib;

namespace MoonlightDashboard.Db
{
    public class Mongo
    {
        static MongoClient? mongoClient = null;
        public static MongoClient MongoClient
        {
            get
            {
                if (mongoClient == null)
                {
                    MongoClientSettings mongoSettings = new ();
                    mongoSettings.Server = new MongoServerAddress(Settings.GetValue<string>("mongo_db_host"), Settings.GetValue<int>("mongo_db_port"));
                    var mongoDbUser = Settings.GetValue<string>("mongo_db_user");
                    var mongoDbPassword = Settings.GetValue<string>("mongo_db_user");
                    var mongoDbAuthSource = Settings.GetValue<string>("mongo_db_user");
                    if (mongoDbUser != null && mongoDbPassword != null && mongoDbAuthSource != null)
                    {
                        mongoSettings.Credential = MongoCredential.CreateCredential(mongoDbAuthSource, mongoDbUser, mongoDbPassword);
                    }
                    mongoClient = new MongoClient(mongoSettings);
                }
                return mongoClient;
            }
        }

        public static IMongoDatabase GetDatabase(string? name = null)
        {
            name ??= Settings.GetValue<string>("mongo_db_db_name");
            return MongoClient.GetDatabase(name);
        }

        public static IMongoCollection<BsonDocument> GetCollection(string name, string? databaseName = null)
        {
            var db = GetDatabase(databaseName);
            return db.GetCollection<BsonDocument>(name);
        }
    }
}
